pkgs/applications/networking/instant-messengers/vk-messenger/default.nix

```diff
@@ -0,0 +1,53 @@
+{ stdenv, fetchurl, dpkg, autoPatchelfHook
+, xorg, gtk2, gnome2, nss, alsaLib
+, udev, libnotify }:
+
+stdenv.mkDerivation rec {
+  name = "vk-messenger-3.9.0";
+  src = {
+    name = "${name}.deb";
+    i686-linux = fetchurl {
+      url = "https://desktop.userapi.com/get_last?platform=linux32&branch=master&packet=deb";
+      sha256 = "0qjzm58vld1lnycmlfvnbmzar4xlxv2qs3hv9al3vgvw9nanji0p";
+      inherit name;
+    };
+    x86_64-linux = fetchurl {
+      url = "https://desktop.userapi.com/get_last?platform=linux64&branch=master&packet=deb";
+      sha256 = "15jxd27drmahqsigrhb3ilx5z447yqyvczm56d6bd83ny1ws1jd7";
+      inherit name;
+    };
+    armv7l-linux = fetchurl {
+      url = "https://desktop.userapi.com/get_last?platform=linux-armv7l&branch=master&packet=deb";
+      sha256 = "1nxgqw3h8fjava4chx3ff2ymjxwdx3nfvw5r9znv73f82sh1gdps";
+      inherit name;
+    };
+  }.${stdenv.system} or (throw "Unsupported system: ${stdenv.system}");
+
+  nativeBuildInputs = [ dpkg autoPatchelfHook ];
+  buildInputs = (with xorg; [
+    libXdamage libXtst libXScrnSaver libxkbfile
+  ]) ++ [
+    gtk2 gnome2.GConf nss alsaLib
+  ];
+  runtimeDependencies = [ udev.lib libnotify ];
+
+  unpackPhase = "dpkg-deb -x $src .";
+
+  installPhase = ''
+    mkdir $out
+    cd usr
+    cp -r --parents bin $out
+    cp -r --parents lib/vk $out
+    cp -r --parents share/applications $out
+    cp -r --parents share/pixmaps $out
+  '';
+
+  meta = with stdenv.lib; {
+    description = "Simple and Convenient Messaging App for VK";
+    homepage = https://vk.com/messenger;
+    license = licenses.free;
```

Comment:
You sure?

Reply:
@dotlambda yeah, this is not right, but VK Messenger doesn't really have a defined license

Reply:
@dotlambda @Yardanico dunno what license it has. Inside the package by share/doc/vk/copyright path there is
```
Copyright (c) 2013-2017 GitHub Inc.

Permission is hereby granted, free of charge, to any person obtaining
a copy of this software and associated documentation files (the
"Software"), to deal in the Software without restriction, including
without limitation the rights to use, copy, modify, merge, publish,
distribute, sublicense, and/or sell copies of the Software, and to
permit persons to whom the Software is furnished to do so, subject to
the following conditions:

The above copyright notice and this permission notice shall be
included in all copies or substantial portions of the Software.

THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
```